btas/generic/cp.h

```diff
@@ -495,21 +502,21 @@ namespace btas{
     /// the \c factor factor matrix with all columns normalized.
     /// \return The column norms of the \c factor factor matrix
 
-    Tensor normCol(int factor) {
-      if(factor >= ndim) BTAS_EXCEPTION("Factor is out of range");
-      auto rank = A[factor].extent(1);
-      auto size = A[factor].size();
+    Tensor normCol(size_t factor) {
+      if (factor >= ndim) BTAS_EXCEPTION("Factor is out of range");
+      ind_t rank = A[factor].extent(1);
+      ord_t size = A[factor].size();
       Tensor lambda(rank);
       lambda.fill(0.0);
       auto A_ptr = A[factor].data();
       auto lam_ptr = lambda.data();
-      for(int i = 0; i < size; ++i){
+      for (ord_t i = 0; i < size; ++i) {
```

Comment:
using overly generic names leads to confusion here and the next 2 loops ... it is OK to use `ij` here and line 519's `for` and `ii` on line 516's `for` ...